src/lib/transition_router/transition_router.ml

```diff
@@ -662,15 +619,27 @@ let run ?(sync_local_state = true) ?(cache_exceptions = false)
           ~directory:persistent_root_location
           ~ledger_depth:(Precomputed_values.ledger_depth precomputed_values)
       in
-      let%map transition_writer_ref =
+      let network_transition_pipe : _ Strict_pipe.Swappable.t =
+        let name = "transition frontier controller pipe" in
+        let drop_f (b_or_h, `Valid_cb valid_cb) =
+          Mina_metrics.(
+            Counter.inc_one Pipe.Drop_on_overflow.router_transitions) ;
+          Mina_block.handle_dropped_transition
+            (block_or_header_to_hash b_or_h)
+            ?valid_cb ~pipe_name:name ~logger
+        in
+        Strict_pipe.Swappable.create ~name
```

Comment:
Could reuse the `create_buffered_pipe` here.

Reply:
It's different:

```ocaml
let create_buffered_pipe ?name ~f () =
  Strict_pipe.create ?name (Buffered (`Capacity 50, `Overflow (Drop_head f)))
```

We could have some common helper for both functions, but I won't make code more readable.